### ISSUE TITLE
Spacy ner scorer return all zeros when using dropout

### ISSUE THREAD

**Issue by mjvallone** [explosion/spaCy#7242]
Hi all,

We are trying to get scores on our training and validation set using Scorer and after a couple of epochs it returns zero for every possible score (f_score, recall, precision and scores per entity type - but NER losses decreases) when **using non-zero dropout**.
If we use zero dropout, it returns non-zero scores as expected.
On our training, we are reducing learning rate if not learning (not getting better scores), changing dropout rate if not learning, increasing epochs if "not learning" (when learning rate has been reduced, we increase epochs by one) , etc.
You can see an example of what "not learning" means talking about dropout.
(Not so) related info: we are using a couple of callbacks to apply configuration settings to our model training.

Here is our code:

```
def change_dropout_fixed(step=0.01, until=0.5):
  """
  [experimental] change the dropout each epoch
  :param step amount to change per epoch
  :param until limit for dropout change  
  """
  def change_dropout_fixed_cb(state, logger, model, optimizer, disabled_pipes):

    if step > 0 and state["dropout"] < until:
      state["dropout"] += step
      logger.info(f"[change_dropout_fixed] touching dropout. New value {state['dropout']}")
    elif step < 0 and state["dropout"] > until:
      # negative step
      state["dropout"] += step
      logger.info(f"[change_dropout_fixed] touching dropout. New value {state['dropout']}")

    else:
      logger.info(f"[change_dropout_fixed] No more room for touching dropout")

    return state

  return change_dropout_fixed_cb

```

I hope someone passed through this because it's kindof blocking us this issue.

## Your Environment
* Operating System: Ubuntu 20.04.2 LTS
* Python Version Used: 3.6.1
* spaCy Version Used: 2.3.2
* Environment Information: -

**Comment by polm** [comment]
Sorry to hear you're having trouble with this. We aren't aware of any existing issues that would cause this - could you give a minimal code example that shows this problem, or some more details about your training setup?

For example, do you have issues using non-zero dropout if you use a fixed value or [one of the built-in schedulers][schedulers]? 

[schedulers]: https://v2.spacy.io/usage/training#tips-dropout
